t1/src/decoder/InstructionDocumentation.scala

```diff
@@ -0,0 +1,425 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2022 Jiuyang Liu <liu@jiuyang.me>
+
+package org.chipsalliance.t1.rtl
+
+import org.chipsalliance.rvdecoderdb.Instruction
+
+/** Generate documentation for each instructions for T1.
+  * The documentation should contain the behavior for instruction in a specific configuration in T1.
+  */
+case class InstructionDocumentation(instruction: Instruction, t1Parameter: T1Parameter) {
```

Comment:
This is how to start to review the decoder: adding documentation to each instructions. Describing the lifetime of each instruction in a specific type of T1.
After finish this review, a clear documentation will be a artifact.

Reply:
we may also add attributes here parameters? I'm not sure.